### PR TITLE
clickhouse-client: echo hint improvements

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Echo queries only after `-- { echo }` (not globally as before). Add `echoOff`/`echoOn` hints.

Detailed description / Documentation draft:
See commit messages.

Cc: @akuzm

### CODE REVIEW

**Comment by akuzm** [comment]
Why do you need to turn echo off? Maybe we can avoid it somehow? This is becoming an entire grammar for echo with three keywords and two different scopes.

**Comment by akuzm** [comment]
I mean, the original `echo` was a just a quick hack because I didn't have time then to just enable it by default for all tests.

**Comment by azat** [comment]
>Why do you need to turn echo off? Maybe we can avoid it somehow? 

To me, most of the time you don't need to echo CREATE TABLE and similar.

File: programs/client/TestHint.h [hunk 1]
```diff
@@ -11,9 +11,35 @@
...
+///
+/// - "-- { serverError 60 }" -- in case of you are expecting server error.
+///
+/// - "-- { clientError 20 }" -- in case of you are expecting client error.
+///
+///   Remember that the client parse the query first (not the server), so for
+///   example if you are expecting syntax error, then you should use
+///   clientError not serverError.
+///
+/// Examples:
```
Comment (alexey-milovidov) [inline programs/client/TestHint.h:26]: I also consider the idea of this PR questionable,
but at least it adds valuable comments - let's merge!

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**